server/license/license.go

```diff
@@ -26,25 +25,25 @@ func IsEnterpriseLicensedOrDevelopment(config *model.Config, license *model.Lice
 }
 
 // isValidSkuShortName returns whether the SKU short name is one of the known strings;
-// namely: E10 or professional, or E20 or enterprise
+// namely: professional, enterprise or enterprise advanced.
 func isValidSkuShortName(license *model.License) bool {
 	if license == nil {
 		return false
 	}
 
 	switch license.SkuShortName {
-	case e10, e20, professional, enterprise:
+	case professional, enterprise, enterpriseAdvanced:
 		return true
 	default:
 		return false
 	}
 }
 
-// IsE10LicensedOrDevelopment returns true when the server is at least licensed with a legacy Mattermost
-// Enterprise E10 License or a Mattermost Professional License, or has `EnableDeveloper` and
+// IsProfessionalLicensedOrDevelopment returns true when the server is at least licensed with
+// a Mattermost Professional License, or has `EnableDeveloper` and
 // `EnableTesting` configuration settings enabled, signaling a non-production, developer mode.
-func IsE10LicensedOrDevelopment(config *model.Config, license *model.License) bool {
-	if IsProfessional(license) || IsEnterprise(license) {
+func IsProfessionalLicensedOrDevelopment(config *model.Config, license *model.License) bool {
```

Comment:
While we're at it, we should fix this:
```suggestion
func IsMinimumProfessionalLicensedOrDevelopment(config *model.Config, license *model.License) bool {
```